Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34826&attempt_id=abfd56b3-a062-4cb7-91c3-69ac5c68f472)

<details><summary>Partial log (click to expand)</summary><p>

```
>               '%.2fms') % (runtime, self.settings.deadline)
            )
E           hypothesis.errors.DeadlineExceeded: Test took 2947.81ms, which exceeds the deadline of 1500.00ms

/nix/store/gck74417yqh4gydgcb5mzkmqqvgmzzsy-python3.6-hypothesis-3.27.0/lib/python3.6/site-packages/hypothesis/core.py:542: DeadlineExceeded
---------------------------------- Hypothesis ----------------------------------
Falsifying example: test_pwhash_str_and_verify(password='     ')
============== 1 failed, 4519 passed, 4 skipped in 219.87 seconds ==============
builder for '/nix/store/n2iwkdx3dblrm63qxp1lr1hkrlg98wbn-python3.6-pynacl-1.2.1.drv' failed with exit code 1
error: build of '/nix/store/5i8zv2nms5019rkdjparpxs6jpjiv7wm-python2.7-pynacl-1.2.1.drv', '/nix/store/n2iwkdx3dblrm63qxp1lr1hkrlg98wbn-python3.6-pynacl-1.2.1.drv' failed
```
</p></details>

--- a/lib/maintainers.nix
+++ b/lib/maintainers.nix
@@ -702,6 +702,7 @@
   utdemir = "Utku Demir <me@utdemir.com>";
   #urkud = "Yury G. Kudryashov <urkud+nix@ya.ru>"; inactive since 2012
   uwap = "uwap <me@uwap.name>";
+  va1entin = "Valentin Heidelberger <github@valentinsblog.com>";
   vaibhavsagar = "Vaibhav Sagar <vaibhavsagar@gmail.com>";
   valeriangalliat = "Valérian Galliat <val@codejam.info>";
   vandenoever = "Jos van den Oever <jos@vandenoever.info>";

--- a/pkgs/development/python-modules/pynacl/default.nix
+++ b/pkgs/development/python-modules/pynacl/default.nix
@@ -1,0 +1,33 @@
+{ stdenv, buildPythonPackage, fetchFromGitHub, pytest, coverage, libsodium, cffi, six, hypothesis}:
+
+buildPythonPackage rec {
+  pname = "pynacl";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "pyca";
+    repo = pname;
+    rev = version;
+    sha256 = "0z9i1z4hjzmp23igyhvg131gikbrr947506lwfb3fayf0agwfv8f";
+  };
+
+  #remove deadline from tests, see https://github.com/pyca/pynacl/issues/370
+  prePatch = ''
+    sed -i 's/deadline=1500, //' tests/test_pwhash.py
+    sed -i 's/deadline=1500, //' tests/test_aead.py
+  '';
+
+  checkInputs = [ pytest coverage hypothesis ];
+  propagatedBuildInputs = [ libsodium cffi six ];
+
+  checkPhase = ''
+    coverage run --source nacl --branch -m pytest
+  '';
+  
+  meta = with stdenv.lib; {
+    maintainers = with maintainers; [ va1entin ];
+    description = "Python binding to the Networking and Cryptography (NaCl) library";
+    homepage = https://github.com/pyca/pynacl/;
+    license = licenses.asl20;
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -20342,23 +20342,7 @@ EOF
     propagatedBuildInputs = with self; [ pynacl six ];
   };
 
-  pynacl = buildPythonPackage rec {
-    name = "pynacl-${version}";
-    version = "0.3.0";
-
-    src = pkgs.fetchurl {
-      url = "mirror://pypi/P/PyNaCl/PyNaCl-0.3.0.tar.gz";
-      sha256 = "1hknxlp3a3f8njn19w92p8nhzl9jkfwzhv5fmxhmyq2m8hqrfj8j";
-    };
-
-    buildInputs = with self; [ pytest coverage ];
-    propagatedBuildInputs = with self; [pkgs.libsodium six cffi pycparser];
-
-    checkPhase = ''
-      coverage run --source nacl --branch -m pytest
-    '';
-
-  };
+  pynacl = callPackage ../development/python-modules/pynacl { };
 
   service-identity = callPackage ../development/python-modules/service_identity { };
 

